File: backend/core/models/signer.py
from django.db import models
from .document import Document

class Signer(models.Model):
    id = models.AutoField(primary_key=True)
    document = models.ForeignKey(Document, on_delete=models.CASCADE)
    token = models.CharField(max_length=255, null=False)
    name = models.CharField(max_length=255, null=False)
    status = models.CharField(max_length=50, null=False)
    email = models.EmailField(max_length=255, null=False)
    externalID = models.CharField(max_length=255, blank=True, null=True)

    def __str__(self):
        return self.name

File: frontend/src/app/core/models/signer.ts
import {PaginatedResponse} from './common';

export interface Signer {
  id: number;
  token: string;
  name: string;
  status: string;
  email: string;
  externalID: string | null;
  document: number;
}

export interface PaginatedSignersResponse extends PaginatedResponse<Signer> {}
